src/_P145_MQxxx.ino

```diff
@@ -0,0 +1,282 @@
+      addUnit(F("Ohm"));
+      addFormTextBox(F("Reference Level"), F("plugin_145_REFLEVEL"), String(PCONFIG_FLOAT_REF), 33);
+      addUnit(F("ppm"));
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data != nullptr)
+      {
+        float calVal = P145_data->getCalibrationValue();
+        if (calVal > 0.0) 
+        {
+          addFormNote((String)F("Current measurement suggests Rzero= ") + (String)calVal);
+        }
+      }
+
+      addFormSeparator(2);
+      // Auto calibration and Temp/hum compensation flags are bitfields in PCONFIG_FLAGS
+      bool compensate = PCONFIG_FLAGS & 0x0001;
+      bool calibrate = (PCONFIG_FLAGS >> 1) & 0x0001;
+      addFormCheckBox(F("Enable automatic calibration"), F("plugin_145_enable_calibrarion"), calibrate);
+      addFormCheckBox(F("Enable temp/humid compensation"), F("plugin_145_enable_compensation"), compensate);
+      addFormNote(F("If this is enabled, the Temperature and Humidity values below need to be configured."));
+      // temperature
+      addHtml(F("<TR><TD>Temperature:<TD>"));
+      addTaskSelect(F("plugin_145_temperature_task"), PCONFIG_TEMP_TASK);
+      LoadTaskSettings(PCONFIG_TEMP_TASK); // we need to load the values from another task for selection!
+      addHtml(F("<TR><TD>Temperature Value:<TD>"));
+      addTaskValueSelect(F("plugin_145_temperature_value"), PCONFIG_TEMP_VAL, PCONFIG_TEMP_TASK);
+      // humidity
+      addHtml(F("<TR><TD>Humidity:<TD>"));
+      addTaskSelect(F("plugin_145_humidity_task"), PCONFIG_HUM_TASK);
+      LoadTaskSettings(PCONFIG_HUM_TASK); // we need to load the values from another task for selection!
+      addHtml(F("<TR><TD>Humidity Value:<TD>"));
+      addTaskValueSelect(F("plugin_145_humidity_value"), PCONFIG_HUM_VAL, PCONFIG_HUM_TASK);
+      LoadTaskSettings(event->TaskIndex); // we need to restore our original taskvalues!
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      PCONFIG_SENSORT   = getFormItemInt(F("plugin_145_sensor"));
+      PCONFIG_FLOAT_RLOAD = getFormItemFloat(F("plugin_145_RLOAD"));
+      PCONFIG_FLOAT_RZERO = getFormItemFloat(F("plugin_145_RZERO"));
+      PCONFIG_FLOAT_REF   = getFormItemFloat(F("plugin_145_REFLEVEL"));
+      bool compensate = isFormItemChecked(F("plugin_145_enable_compensation") );
+      bool calibrate  = isFormItemChecked(F("plugin_145_enable_calibrarion") );
+      PCONFIG_FLAGS = compensate + (calibrate << 1);
+      PCONFIG_TEMP_TASK = getFormItemInt(F("plugin_145_temperature_task"));
+      PCONFIG_TEMP_VAL  = getFormItemInt(F("plugin_145_temperature_value"));
+      PCONFIG_HUM_TASK  = getFormItemInt(F("plugin_145_humidity_task"));
+      PCONFIG_HUM_VAL   = getFormItemInt(F("plugin_145_humidity_value"));
+
+      P145_data_struct *P145_data = new (std::nothrow) P145_data_struct();
+      initPluginTaskData(event->TaskIndex, P145_data);
```

Comment:
Do not init the plugin on saving.
When a task is saved, the `PLUGIN_EXIT` will be called, followed by `PLUGIN_INIT` (when enabled)
So there is no need to init anything when saving it.
If there is something special where `PLUGIN_EXIT` should be called before `PLUGIN_SAVE` or after, then you can set `Device[deviceCount].ExitTaskBeforeSave = false;` (or `true`) in the `PLUGIN_DEVICE_ADD` call.

Reply:
Thanks, copy-paste error. Intended was to access the current P145_data_struct